### DISCUSSION TITLE
Avalonia vs WinUI popularity

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#8144] [General]
Some had concerns about WinUI vs Avalonia. Here's something I just saw.

Search "mvvm blazor" in StackOverflow. 62 results.

"mvvm avalonia": 39 results.

"mvvm winui": 18 results.

Anyone uses WinUI at all? Overall, none of the latest .NET platforms seem overly popular.

**Comment by robloo** [comment]
I was active in WinUI repo. However, my company moved on. Reasons are here: https://github.com/microsoft/microsoft-ui-xaml/issues/3639

Microsoft is repeating the same mistakes as the last few UI frameworks (UWP, Silverlight) as far as I'm concerned. Their rate of change is ridiculously slow now too. Serious issues have been open with no movement for years. WinUI 3 was never open sourced.

Maui is also not what several developers have been asking for: https://github.com/dotnet/maui/issues/43

Avalonia is what devs has always wanted and asked for. The only thing holding it back was Microsoft didn't start it. The technology in Avalonia now speaks for itself though. It is the .NET/C# response to flutter (actually was first by 5 years). It is what UWP should have been.

**Reply by mysteryx93** [reply]
I hear ya... the sad truth is that desktop development in .NET is pretty much dead because people don't know what platform to use. WPF never was important enough to make it worth porting to MONO, and WinForms is way too old for new projects, and anything newer than WPF is lackbuster at best.

I've seen very few desktop applications that use .NET

Since most companies have moved away from .NET for desktop apps, how many remain to even look into Avalonia? Perhaps a bunch will come back, but most will have found a home elsewhere.

When I try to apply MVVM to modern frameworks, it's just a bit annoying how little there is for MVVM Blazor or MVVM WinUI. It's like nobody cares. Although Blazor does get some interest.

**Reply by robloo** [reply]
There are lots of special-purpose LOB applications still using .NET and WinForms/WPF. Companies still use .NET all over in various industries. We don't usually see these though and it's true consumer-facing apps have migrated almost entirely to web-based framework.

After watching the .NET UI space for a few decades, I'm fairly confident Avalonia is going to break through where others have failed in the past. It is everything developers have asked for and it also isn't encumbered by Microsoft's bureaucracy. Microsoft kills a lot of good ideas themselves with leadership/management changes and giving up as soon as the first roadblocks are found. Avalonia is past all this.

.NET UI future is actually very bright. I agree this is all 10 years too late. But now we have Maui, Uno and Avalonia that provide cross-platform UI to keep .NET relevant. Software is also cyclical and developers will probably re-discover the benefits of native apps again and start moving to solutions like .NET/Avalonia. Avalonia is also moving to run on the web as well. Bottom line, .NET looked like it was on the last breath then Xamarin acquisition happened and these new UI frameworks like Avalonia are powerful enough to be useful. Companies with LOB applications will seriously consider moving to Avalonia. There are many examples of this already. Even consumer-facing apps are taking a good look and choosing Avalonia as an alternative to Flutter (.NET/C# is better than Dart).

**Reply by mysteryx93** [reply]
On a side note, I just published 2 Avalonia software to the Arch Linux AUR
https://github.com/mysteryx93/NaturalGroundingPlayer/wiki/432hz-Player
https://github.com/mysteryx93/NaturalGroundingPlayer/wiki/Powerliminals-Player

Indeed, I think .NET desktop apps running on Linux is pretty unheard of, and people will take notice of how potent it is.
